### PR TITLE
[WIP] 🚧 🐛 Fix multiple highlights for the open diagram in the SolutionExplorer

### LINKED ISSUE #1212: 🐛 Open diagram is highlighted in the SolutionExplorer twice, when the FileSystem Solution is still open
#### Describe your issue

When deploying a Diagram to the ProcessEngine or open a Solution which contains a diagram which has the same name, as a deployed Diagram, both entries are highlighted as open in the SolutionExplorer.

screenshot 2018-12-20 at 09 57 35

#### How can others reproduce the issue?

* Open a Solution
* Deploy a Diagram
* See, that the diagram is highlighted twice in the SolutionExplorer. 

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by robpal94**
---

**Changes:**

1. Fixes a bug, where multiple Entries of the SolutionExplorer where highlighted, if multiple open solutions contained a diagram with the same name. 

**Issues:**

Closes #1212 

PR: #1215

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [ ] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [ ] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.html [hunk 1]
```diff
@@ -38,7 +38,9 @@
           displayed-solution-entry.bind="solutionEntry"
           solution-service.bind="solutionEntry.service"
           solution-is-single-diagrams.bind="solutionEntry.isSingleDiagramService"
-          view-model.ref="solutionEntryViewModels[solutionEntry.uri]">
+          current-opened-root-solution-uri.bind="currentOpenedRootSolutionUri"
```
Comment (Vyperus) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.html]: So what is a root solution uri? And what makes it root?

Comment (robpal94) [reply src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.html]: It's the origin of the solution, where the diagram is located. But yeah, you are right. `root` is a bit misleading here (its basically the `root` path of the solution, where the diagram is located).

**Comment by Vyperus** [comment]
@Paulomart @robpal94 You can actually do that now, all information you need to determine the active diagram and solution are in the uri

**Comment by robpal94** [comment]
@Paulomart 
> Why does the the solution-explorer-solution needs to keep track which diagram is currenlty active? 

This is also done so in the current implementation. See: https://github.com/process-engine/bpmn-studio/blob/develop/src/modules/solution-explorer/solution-explorer-solution/solution-explorer-solution.ts#L755

This Pr was more meant to be a quickfix instead of a complete solution for this Problem. But I guess you are right, using stuff like the  `click location` to determine the exact uri of the opened diagram is just to janky. 

We need a global instance which tells each solution, which solution should be active. I kinda likes the idea of the `getActiveSolution()` Method of the `SolutionExplorerService` but it got removed.

I think this needs further discussion, but we should not spend to much time on it because imo, the issue is kinda annoying.

**Comment by robpal94** [comment]
Oh wow, im ***sooooo stupid*** 🤦‍♀️
I didn't see that we actually pass the uri with the RouteParameters, where it can accessed via `this._router.currentInstruction.queryParams.solutionUri`. Sorry for the disturbance 😇

**Pull request closed without merging**